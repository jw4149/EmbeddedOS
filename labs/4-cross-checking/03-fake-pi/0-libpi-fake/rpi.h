#ifndef __RPI_H__
#define __RPI_H__
// simplest way to rewrap --- just define replacements in header --- don't
// even need .c
//
// easy, but two downsides: 
//   1. we are polluting the pi program namespace with system includes.
//   2. we aren't using the raw rpi.h so can get different behavior.
#include <stdio.h>
#include <stdlib.h>

// redefine printk as fprintf using variadic macros.
#define printk(args...) fprintf(stderr, args)

static void inline clean_reboot(void) {
    printk("DONE!!!\n");
    exit(0);
}

void notmain(void);
#endif
