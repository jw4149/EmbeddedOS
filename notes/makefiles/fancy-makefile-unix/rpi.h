#include <stdio.h>

#define printk printf
#define debug printf
