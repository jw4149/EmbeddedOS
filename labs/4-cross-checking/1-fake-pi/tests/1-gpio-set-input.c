#include "rpi.h"
void notmain(void) {
    gpio_set_input(20);
}
