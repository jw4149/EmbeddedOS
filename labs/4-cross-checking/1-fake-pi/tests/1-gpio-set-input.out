TRACE:0: calling pi code
TRACE:1: GET32(0x20200008) = 0x66334873
TRACE:2: PUT32(0x20200008) = 0x66334870
TRACE:3: GET32(0x20200008) = 0x66334870
TRACE:4: PUT32(0x20200008) = 0x66334870
TRACE:5: GET32(0x20200008) = 0x66334870
TRACE:6: PUT32(0x20200008) = 0x66334870
TRACE:7: GET32(0x20200008) = 0x66334870
TRACE:8: PUT32(0x20200008) = 0x66334870
TRACE:9: GET32(0x20200008) = 0x66334870
TRACE:10: PUT32(0x20200008) = 0x66334870
TRACE:11: GET32(0x20200008) = 0x66334870
TRACE:12: PUT32(0x20200008) = 0x66334870
TRACE:13: GET32(0x20200008) = 0x66334870
TRACE:14: PUT32(0x20200008) = 0x66334870
TRACE:15: GET32(0x20200008) = 0x66334870
TRACE:16: PUT32(0x20200008) = 0x66334870
TRACE:17: GET32(0x20200008) = 0x66334870
TRACE:18: PUT32(0x20200008) = 0x66334870
TRACE:19: GET32(0x20200008) = 0x66334870
TRACE:20: PUT32(0x20200008) = 0x66334870
TRACE:21: GET32(0x20200008) = 0x66334870
TRACE:22: PUT32(0x20200008) = 0x66334870
TRACE:23: GET32(0x20200008) = 0x66334870
TRACE:24: PUT32(0x20200008) = 0x66334870
TRACE:25: GET32(0x20200008) = 0x66334870
TRACE:26: PUT32(0x20200008) = 0x66334870
TRACE:27: GET32(0x20200008) = 0x66334870
TRACE:28: PUT32(0x20200008) = 0x66334870
TRACE:29: GET32(0x20200008) = 0x66334870
TRACE:30: PUT32(0x20200008) = 0x66334870
TRACE:31: GET32(0x20200008) = 0x66334870
TRACE:32: PUT32(0x20200008) = 0x66334870
TRACE:33: GET32(0x20200008) = 0x66334870
TRACE:34: PUT32(0x20200008) = 0x66334870
TRACE:35: GET32(0x20200008) = 0x66334870
TRACE:36: PUT32(0x20200008) = 0x66334870
TRACE:37: GET32(0x20200008) = 0x66334870
TRACE:38: PUT32(0x20200008) = 0x66334870
TRACE:39: GET32(0x20200008) = 0x66334870
TRACE:40: PUT32(0x20200008) = 0x66334870
TRACE:41: GET32(0x20200008) = 0x66334870
TRACE:42: PUT32(0x20200008) = 0x66334870
TRACE:43: GET32(0x20200008) = 0x66334870
TRACE:44: PUT32(0x20200008) = 0x66334870
TRACE:45: GET32(0x20200008) = 0x66334870
TRACE:46: PUT32(0x20200008) = 0x66334870
TRACE:47: GET32(0x20200008) = 0x66334870
TRACE:48: PUT32(0x20200008) = 0x66334870
TRACE:49: GET32(0x20200008) = 0x66334870
TRACE:50: PUT32(0x20200008) = 0x66334870
TRACE:51: GET32(0x20200008) = 0x66334870
TRACE:52: PUT32(0x20200008) = 0x66334870
TRACE:53: GET32(0x20200008) = 0x66334870
TRACE:54: PUT32(0x20200008) = 0x66334870
TRACE:55: GET32(0x20200008) = 0x66334870
TRACE:56: PUT32(0x20200008) = 0x66334870
TRACE:57: GET32(0x20200008) = 0x66334870
TRACE:58: PUT32(0x20200008) = 0x66334870
TRACE:59: GET32(0x20200008) = 0x66334870
TRACE:60: PUT32(0x20200008) = 0x66334870
TRACE:61: GET32(0x20200008) = 0x66334870
TRACE:62: PUT32(0x20200008) = 0x66334870
TRACE:63: GET32(0x20200008) = 0x66334870
TRACE:64: PUT32(0x20200008) = 0x66334870
TRACE:65: GET32(0x20200008) = 0x66334870
TRACE:66: PUT32(0x20200008) = 0x66334871
TRACE:67: GET32(0x20200008) = 0x66334871
TRACE:68: PUT32(0x20200008) = 0x66334871
TRACE:69: GET32(0x20200008) = 0x66334871
TRACE:70: PUT32(0x20200008) = 0x66334871
TRACE:71: GET32(0x20200008) = 0x66334871
TRACE:72: PUT32(0x20200008) = 0x66334871
TRACE:73: GET32(0x20200008) = 0x66334871
TRACE:74: PUT32(0x20200008) = 0x66334871
TRACE:75: GET32(0x20200008) = 0x66334871
TRACE:76: PUT32(0x20200008) = 0x66334871
TRACE:77: GET32(0x20200008) = 0x66334871
TRACE:78: PUT32(0x20200008) = 0x66334871
TRACE:79: GET32(0x20200008) = 0x66334871
TRACE:80: PUT32(0x20200008) = 0x66334871
TRACE:81: GET32(0x20200008) = 0x66334871
TRACE:82: PUT32(0x20200008) = 0x66334871
TRACE:83: GET32(0x20200008) = 0x66334871
TRACE:84: PUT32(0x20200008) = 0x66334871
TRACE:85: GET32(0x20200008) = 0x66334871
TRACE:86: PUT32(0x20200008) = 0x66334871
TRACE:87: GET32(0x20200008) = 0x66334871
TRACE:88: PUT32(0x20200008) = 0x66334871
TRACE:89: GET32(0x20200008) = 0x66334871
TRACE:90: PUT32(0x20200008) = 0x66334871
TRACE:91: GET32(0x20200008) = 0x66334871
TRACE:92: PUT32(0x20200008) = 0x66334871
TRACE:93: GET32(0x20200008) = 0x66334871
TRACE:94: PUT32(0x20200008) = 0x66334871
TRACE:95: GET32(0x20200008) = 0x66334871
TRACE:96: PUT32(0x20200008) = 0x66334871
TRACE:97: GET32(0x20200008) = 0x66334871
TRACE:98: PUT32(0x20200008) = 0x66334871
TRACE:99: GET32(0x20200008) = 0x66334871
TRACE:100: PUT32(0x20200008) = 0x66334871
TRACE:101: GET32(0x20200008) = 0x66334871
TRACE:102: PUT32(0x20200008) = 0x66334871
TRACE:103: GET32(0x20200008) = 0x66334871
TRACE:104: PUT32(0x20200008) = 0x66334871
TRACE:105: GET32(0x20200008) = 0x66334871
TRACE:106: PUT32(0x20200008) = 0x66334871
TRACE:107: GET32(0x20200008) = 0x66334871
TRACE:108: PUT32(0x20200008) = 0x66334871
TRACE:109: GET32(0x20200008) = 0x66334871
TRACE:110: PUT32(0x20200008) = 0x66334871
TRACE:111: GET32(0x20200008) = 0x66334871
TRACE:112: PUT32(0x20200008) = 0x66334871
TRACE:113: GET32(0x20200008) = 0x66334871
TRACE:114: PUT32(0x20200008) = 0x66334871
TRACE:115: GET32(0x20200008) = 0x66334871
TRACE:116: PUT32(0x20200008) = 0x66334871
TRACE:117: GET32(0x20200008) = 0x66334871
TRACE:118: PUT32(0x20200008) = 0x66334871
TRACE:119: GET32(0x20200008) = 0x66334871
TRACE:120: PUT32(0x20200008) = 0x66334871
TRACE:121: GET32(0x20200008) = 0x66334871
TRACE:122: PUT32(0x20200008) = 0x66334871
TRACE:123: GET32(0x20200008) = 0x66334871
TRACE:124: PUT32(0x20200008) = 0x66334871
TRACE:125: GET32(0x20200008) = 0x66334871
TRACE:126: PUT32(0x20200008) = 0x66334871
TRACE:127: GET32(0x20200008) = 0x66334871
TRACE:128: PUT32(0x20200008) = 0x66334871
TRACE: exiting after 129 trace statements
TRACE: pi exited cleanly: 7 calls to random
